### PR TITLE
cephadm: Add "--format" option to "ceph orch status"

### PR DESCRIPTION
**Pull request by ricardoasmarques**
With this PR, it's now possible to specify `--format` on `ceph orch status` command, e.g.:

``` 
master:~ # ceph orch status
Backend: cephadm
Available: False (SSH keys not set. Use `ceph cephadm set-priv-key` and `ceph cephadm set-pub-key` or `ceph cephadm generate-key`)
``` 
``` 
master:~ # ceph orch status --format=json

{"available": false, "backend": "cephadm", "reason": "SSH keys not set. Use `ceph cephadm set-priv-key` and `ceph cephadm set-pub-key` or `ceph cephadm generate-key`"}
``` 
``` 
master:~ # ceph orch status --format=json-pretty

{
  "available": false,
  "backend": "cephadm",
  "reason": "SSH keys not set. Use `ceph cephadm set-priv-key` and `ceph cephadm set-pub-key` or `ceph cephadm generate-key`"
}
``` 
``` 
master:~ # ceph orch status --format=yaml
available: false
backend: cephadm
reason: SSH keys not set. Use `ceph cephadm set-priv-key` and `ceph cephadm set-pub-key`
  or `ceph cephadm generate-key`
``` 

Fixes: https://tracker.ceph.com/issues/46233

Signed-off-by: Ricardo Marques <rimarques@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/orchestrator/module.py [hunk 1]
```diff
@@ -1241,21 +1241,31 @@ def _cancel(self):
...
-            return HandleCommandResult(stdout="Backend: {0}\nAvailable: {1}{2}".format(
-                                           o, avail,
-                                           " ({0})".format(why) if not avail else ""
-                                       ))
+            output = "Backend: {0}".format(result['backend'])
+            if 'available' in result:
+                output += "\nAvailable: {0}".format(result['available'])
+                if 'reason' in result:
+                    output += ' ({0})'.format(result['reason'])
+        return HandleCommandResult(stdout=output)
```
Comment (sebastian-philipp) [inline src/pybind/mgr/orchestrator/module.py:1268]:
```suggestion
        return HandleCommandResult(retval=0 if avail else EINVAL, stdout=output)
```

**Comment by sebastian-philipp** [comment]
If you want, you can add a call to `ceph orch status --format ...` in https://github.com/ceph/ceph/blob/749da74e2d9af075c2f365516cf9ea4370b58ec0/qa/tasks/cephadm_cases/test_cli.py to verify this.

File: src/pybind/mgr/orchestrator/module.py (hunk 1 above)
Comment (ricardoasmarques) [reply src/pybind/mgr/orchestrator/module.py:1268]: If we return a non zero exit code, we will always see the `Error EINVAL:` prefix, which for instance, makes the following output an invalid json:

```
# ceph orch status --format=json
Error EINVAL: {"available": false, "backend": "cephadm", "reason": "SSH keys not set. Use `ceph cephadm set-priv-key` and `ceph cephadm set-pub-key` or `ceph cephadm generate-key`"}
```

For this reason, let's keep exit code zero for now.

**Comment by ricardoasmarques** [comment]
> If you want, you can add a call to `ceph orch status --format ...` in https://github.com/ceph/ceph/blob/749da74e2d9af075c2f365516cf9ea4370b58ec0/qa/tasks/cephadm_cases/test_cli.py to verify this.

Done

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2020-07-01_09:26:21-rados:cephadm-wip-swagner-testing-2020-07-01-0956-distro-basic-smithi/

**Pull request merged**